src/cmd/run.js

```diff
@@ -9,15 +9,60 @@ import {createLogger} from '../util/logger';
 import getValidatedManifest, {getManifestId} from '../util/manifest';
 import defaultSourceWatcher from '../watcher';
 
+const log = createLogger(__filename);
+
+// Flow types
+
 // Import objects that are only used as Flow types.
 import type FirefoxProfile from 'firefox-profile';
+import type {OnSourceChangeFn} from '../watcher';
+import type Watchpack from 'watchpack';
+import type {
+  FirefoxProcess,
+} from '../firefox/index';
+import type {
+  FirefoxConnectorFn, RemoteFirefox,
+  FirefoxRDPResponseAddon,
+} from '../firefox/remote';
+import type {ExtensionManifest} from '../util/manifest';
 
-const log = createLogger(__filename);
+export type WatcherCreatorParams = {
+  addonId: string,
+  client: RemoteFirefox,
+  sourceDir: string,
+  artifactsDir: string,
+  onSourceChange?: OnSourceChangeFn,
+};
+
+export type ReloadStrategyParams = {
```

Comment:
I'd prefer to see this directly above the [defaultReloadStrategy()](https://github.com/rpl/web-ext/blob/d5083f42002b2dcb5e72f8c886997d0035c1c0c7/src/cmd/run.js#L79) function.